Sources/SmithyHTTPAPI/HTTPResponse.swift

```diff
@@ -8,43 +8,48 @@
 import protocol Smithy.ResponseMessage
 import protocol Smithy.Stream
 import enum Smithy.ByteStream
-import class Foundation.DispatchQueue
+import class Foundation.NSRecursiveLock
 
-public class HTTPResponse: HTTPURLResponse, ResponseMessage {
+public final class HTTPResponse: ResponseMessage, @unchecked Sendable {
+    private var lock = NSRecursiveLock()
```

Comment:
The `headers`, `body`, and `statusCode` properties on `HTTPResponse` are locked with a recursive lock.

(A DispatchQueue was previously used on `statusCode`; it has been converted to `NSRecursiveLock` for simplicity.)